From ef1fa9bef497d9491a51017f3cad0135a23bb0b4 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Tue, 20 Aug 2019 17:33:15 +0800
Subject: [PATCH] readline.c: fix cannot get history

if history_offset not increate when add history,
it will make current history event not align with offset,
and cannot get history correctly.

Upstream-Status: Submitted [tech-userlevel@NetBSD.org] 

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 src/readline.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/readline.c b/src/readline.c
index 43ac5bd..54b0548 100644
--- a/src/readline.c
+++ b/src/readline.c
@@ -1476,7 +1476,10 @@ add_history(const char *line)
 	if (ev.num == history_length)
 		history_base++;
 	else
+        {
+                history_offset++;
 		history_length = ev.num;
+        }
 	return 0;
 }
 
-- 
2.7.4

